mesonbuild/dependencies/dev.py

```diff
@@ -506,6 +506,161 @@ def _original_module_name(self, module: str) -> str:
         return module
 
 
+class ClangSystemDependency(SystemDependency):
+
+    def __init__(self, name: str, env: Environment, kwargs: T.Dict[str, T.Any], language: T.Optional[str] = None) -> None:
+        # Default to the C++ bindings, as they are more powerful and more
+        # commonly used than the C bindings.
+        language = kwargs.get('language', language)
+        if language not in {None, 'c', 'cpp'}:
+            raise DependencyException('Clang only provides C and C++ language support')
+
+        super().__init__(name, env, kwargs, language)
+        self.feature_since = ('1.5.0', '')
+        self.module_details: T.List[str] = []
+
+        # Clang may be installed a number of different ways:
+        #
+        # 1. Clang is installed directly in a common search path
+        # 2. Clang is installed alongside LLVM in a separate path to allow multiple versions
+        #    to be co-installed. (Debian and Gentoo do this)
+        # 3. LLVM and Clang are installed in separate, default search paths. (NixOS does this)
+        #
+        # In order to accommodate all three of these we need to search both in
+        # the LLVM directory and outside of it. Start with the LLVM dir to avoid
+        # a situation where there is Clang next to LLVM and a different one in a
+        # common path
+        #
+        # Try to handle the combinations of CMake and config-tool LLVM with this
+        # method, even though it probably doesn't make sense to use the system
+        # finder for Clang with CMake LLVM
+        llvm = T.cast('T.Optional[ExternalDependency]', kwargs.get('llvm'))
+        if llvm is not None:
+            if not llvm.found():
+                mlog.debug('Passed LLVM was not found, treating Clang as not found')
+                return
+            if self.version_reqs and not mesonlib.version_compare_many(llvm.version, self.version_reqs):
+                mlog.debug('Passed LLVMs version does not match the version required for Clang, treating it as not found')
+                return
+            self.ext_deps.append(llvm)
+        else:
+            if not self._add_sub_dependency(
+                llvm_factory(
+                    env, self.for_machine, {'required': False, 'version': kwargs.get('version'), 'method': 'config-tool'})):
+                return
+            llvm = T.cast('ExternalDependency', self.ext_deps[0])
+        # Clang and LLVM need to have the same version
+        self.version = llvm.version
+
+        modules = stringlistify(extract_as_list(kwargs, 'modules'))
+        if not modules and language == 'cpp':
+            mlog.warning('Clang dependency without modules works correctly for dynamically linked Clang, '
```

Comment:
casing on 'clang' seems inconsistent

Reply:
Perhaps a comment here would be good because I am not understanding the language check

Reply:
I've added a comment here, let me know if that clears things up.